azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -201,15 +215,44 @@ private boolean isMultiExecutorMode() {
    */
   private void refreshExecutors() {
     synchronized (activeExecutors) {
-      // TODO: rest api call to refresh executor stats
+      ExecutorService taskExecutors =
+        Executors.newFixedThreadPool(activeExecutors.size());
+      for (final Executor executor : activeExecutors) {
+
+        // execute each executorInfo refresh task
+        taskExecutors.execute(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              String jsonResponse =
+                callExecutorForJsonString(executor.getHost(),
+                  executor.getPort(), "/serverstastics", null);
+              executor.setExecutorInfo(ExecutorInfo
+                .fromJSONString(jsonResponse));
+            } catch (Exception e) {
+              logger.error("Failed to update ExecutorInfo executorId :"
+                + executor, e);
```

Comment:
Good catch, I think this change got dropped as it was conflicting for #478. Modified toString method.